documentation/architecture/decisions/0006-deploying-prometheus-to-private-networks-in-aws.md

```diff
@@ -0,0 +1,232 @@
+end up having to consider more VPC Peering connections in their
+security analyses, and this doesn't feel like a particularly scalable
+way for us to offer services to client teams.
+
+Finally, we believe that VPC Peering is something that has to be
+accepted manually on the receiving account console (at least when
+peering across AWS accounts), which compounds the scaling problem.
+
+There are two sub cases worth exploring here:
+
+#### A single prometheus VPC peers with multiple client VPCs
+
+In this model, we would build a single prometheus service in a VPC
+owned by us, and it would have VPC peering arrangements with multiple
+client team VPCs in order to scrape metrics from each of them.
+
+This has the benefit that we can run fewer instances to scrape metrics
+from the same number of environments.
+
+This has some drawbacks:
+
+ - the single VPC becomes more privileged, because it has access to
+   more environments.  This means that a compromise of the prometheus
+   VPC could lead to a compromise of more clients' VPCs.
+
+#### A single prometheus VPC peers with only a single client VPC
+
+In this scenario, we would build a single prometheus in its own VPC
+for each client team VPC we offer the service to.
+
+This avoids some of the drawbacks of the previous case, in that the
+prometheus doesn't have privileges to access multiple separate VPCs.
+
+[VPC Peering]: https://docs.aws.amazon.com/AmazonVPC/latest/PeeringGuide/Welcome.html
+
+### Use VPC Endpoint Services to access scrape targets
+
+This is a similar idea to VPC Peering.  [VPC Endpoint Services][] (aka
+AWS PrivateLink) provides a way to provide services to a VPC, again
+potentially in another account.
+
+This allows you to make a single Network Load Balancer (NLB) appear
+directly available (ie without going through a NAT Gateway) in another
+VPC.
+
+In the case of Prometheus, because it has a pull model, it seems
+likely that the only way we could make this work would be by having
+the client team provide the endpoint service and prometheus consume
+it.  This would mean the client team would need to add a routing layer
+(possibly path- or vhost-based routing, possibly using an ALB) to
+distribute scrape requests to individual scrape targets.
+
+This has the following advantages:
+
+ - the IP address spaces in the prometheus VPC and the client team VPC
+   are completely independent
+
+However it has some drawbacks:
+
+ - it feels like we're not using Endpoint Services in a designed use
+   case. in other words, it feels like a bit of a hack.
+ - Prometheus is designed to be close to its targets, so that there
+   are fewer things to go wrong and prevent scraping.  The more layers
+   of routing between prometheus and its scrape targets, the more
+   chance we'll lose metrics during an outage, exactly when we need
+   them.
+
+[VPC Endpoint Services]: https://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/endpoint-service.html
+
+## Decision
+
```

Comment:
Might be nice to make it explicit and refer back to the previous headings i.e. `Client team provides IAM access so that we can deploy prometheus ourselves` as the decision section at the moment makes no explicit mention of IAM access